File: hangman.java
import java.util.Scanner;

public class HangmanGame {
    private static final String[] WORDS = { "java", "python", "programming", "hangman", "computer" };
    private static final int MAX_TRIES = 6;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        boolean playAgain = true;

        while (playAgain) {
            String secretWord = selectRandomWord(WORDS);
            String guessedWord = initializeGuessedWord(secretWord);
            int tries = 0;
            boolean isGameOver = false;

            System.out.println("Welcome to Hangman!");
            System.out.println("Guess the word:");

            while (!isGameOver) {
                displayHangman(tries);
                displayWord(guessedWord);

                char guess = getGuess(scanner);

                if (secretWord.contains(String.valueOf(guess))) {
                    guessedWord = updateGuessedWord(secretWord, guessedWord, guess);
                } else {
                    tries++;
                    System.out.println("Incorrect guess. Tries left: " + (MAX_TRIES - tries));
                }

                if (guessedWord.equals(secretWord) || tries >= MAX_TRIES) {
                    isGameOver = true;
                    if (guessedWord.equals(secretWord)) {
                        System.out.println("Congratulations! You guessed the word: " + secretWord);
                    } else {
                        System.out.println("Game over! The word was: " + secretWord);
                    }
                }
            }

            System.out.print("Play again? (yes/no): ");
            String playAgainInput = scanner.next().toLowerCase();
            playAgain = playAgainInput.equals("yes");
        }

        System.out.println("Thanks for playing Hangman!");
        scanner.close();
    }

    private static String selectRandomWord(String[] words) {
        int randomIndex = (int) (Math.random() * words.length);
        return words[randomIndex];
    }

    private static String initializeGuessedWord(String word) {
        StringBuilder builder = new StringBuilder(word.length());
        for (int i = 0; i < word.length(); i++) {
            builder.append("_");
        }
        return builder.toString();
    }

    private static void displayHangman(int tries) {
        // Implement hangman drawing here
    }

    private static void displayWord(String word) {
        System.out.println("Word: " + word);
    }

    private static char getGuess(Scanner scanner) {
        System.out.print("Enter a letter: ");
        String input = scanner.next().toLowerCase();
        if (input.length() != 1 || !Character.isLetter(input.charAt(0))) {
            System.out.println("Invalid input. Please enter a single letter.");
            return getGuess(scanner);
        }
        return input.charAt(0);
    }

    private static String updateGuessedWord(String secretWord, String guessedWord, char guess) {
        StringBuilder builder = new StringBuilder(guessedWord);
        for (int i = 0; i < secretWord.length(); i++) {
            if (secretWord.charAt(i) == guess) {
                builder.setCharAt(i, guess);
            }
        }
        return builder.toString();
    }
}

File: Hangman.py
import random

def hangman():
    words = ['python', 'hangman', 'coding', 'game', 'player', 'guess', 'word', 'computer']
    secret_word = random.choice(words)
    word_length = len(secret_word)
    guessed_letters = []
    attempts = 6

    print("Welcome to Hangman!")
    print(f"The word contains {word_length} letters.")
    print("You have 6 attempts to guess the word.")

    while attempts > 0:
        display = ''
        for letter in secret_word:
            if letter in guessed_letters:
                display += letter + ' '
            else:
                display += '_ '

        print(display)

        if '_' not in display:
            print("Congratulations! You've guessed the word.")
            break

        guess = input("Guess a letter: ").lower()

        if len(guess) != 1 or not guess.isalpha():
            print("Please enter a single letter.")
            continue

        if guess in guessed_letters:
            print("You've already guessed that letter. Try a different one.")
            continue

        guessed_letters.append(guess)

        if guess not in secret_word:
            attempts -= 1
            print(f"Sorry, '{guess}' is not in the word. Attempts left: {attempts}")
        else:
            print("Good guess!")

    if '_' in display:
        print(f"Sorry, you've run out of attempts. The word was: {secret_word}")

if __name__ == "__main__":
    hangman()
